### DISCUSSION TITLE
Help migrating v2 options to v3 swagger_ui_options

### DISCUSSION THREAD

**Discussion by alanbchristie** [spec-first/connexion#1963] [Q&A]
I'm having more difficulty than I expected migrating from v2 to v3. I'm now in a mode of "random button pushing" with regard to the configuration to see if I can get the swagger and API available on the server, but my initial attempts have failed. I've tried numerous variations of the v3 configuration but I'm constantly met with `The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.`.

Can anyone provide some help on what I'm missing?

## v2 (2.14.2)
My v2 code essentially does this...
```
CONNEXION_OPTIONS: Dict[str, Any] = {
    'swagger_url': '/api',
    'swagger_ui_config': {'oauth2RedirectUrl': _OAUTH2_REDIRECT_URL},
}
connexion_app = connexion.FlaskApp(
    __name__, specification_dir='openapi/', options=CONNEXION_OPTIONS
)
```

My OpenAPI file is called `openapi.yaml` and can be found in the `specification_dir` (relative to the app), e.g. `openapi/openapi.yaml`. I get my swagger at the URL I expect i.e. `https://example.com/api`

## v3 (3.1.0)
My various attempts at getting the v3 code to work result in `werkzeug.exceptions` value of `The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.`

Given my v2 `connexion.flaskApp` initialisation object, what would a v3 object look like?

**Comment by alanbchristie** [comment]
Having read the discussion "[Upgrade from 2.6.0 to 3.0.1, and all APIs are unavailable](https://github.com/spec-first/connexion/discussions/1795)" I suspect that the configuration may look like this, with the use of a `SwaggerUIOptions` object: -

```
swagger_ui_options = SwaggerUIOptions(
    swagger_ui_path='/api',
    swagger_ui_config={'oauth2RedirectUrl': _OAUTH2_REDIRECT_URL},
)
connexion_app = connexion.FlaskApp(
    __name__,
    specification_dir="openapi",
    swagger_ui_options=swagger_ui_options,
)
```

I can then use the `add_api()` method to attach the API: -

```
connexion_app.add_api(
    'openapi.yaml',
    strict_validation=True,
    validate_responses=True,
)
```

This, at least, gets me to unit testing, which is further complicated from the switch to `httpx` from `requests` (maybe more on that later).

**Comment by chrisinmtown** [comment]
I realize you posted in August and now it's November, so you're probably long past this bump. I saw that messsage when my code was calling the wrong app. Connection v3 requires you to run the Connexion app, not the Flask app. HTH

**Comment by RobbeSneyders** [comment]
Indeed, as discussed in https://github.com/spec-first/connexion/discussions/1795, this is due to running the Flask instead of Connexion app.
